oak_functions/loader/src/server.rs

```diff
@@ -0,0 +1,118 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::Context;
+use http::{request::Request, response::Response};
+use hyper::{
+    service::{make_service_fn, service_fn},
+    Body, Server, StatusCode,
+};
+use log::info;
+use std::{fs, net::SocketAddr, sync::Arc};
+
+const MAIN_FUNCTION_NAME: &str = "main";
+
+/// An HTTP server that serves the `main` function of a wasm module on each invocation.
+pub struct WasmServer {
+    /// Server address to listen for client requests on.
+    address: SocketAddr,
+    /// The wasm engine that runs the given wasm module per incoming request.
```

Comment:
```suggestion
    /// The Wasm engine that runs the given wasm module per incoming request.
```

and elsewhere (see https://webassembly.org/)